File: cpp/1695.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;

class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        set<int> s;
        int ans = 0, cur = 0;
        int l = 0, r = 0;
        int len = nums.size();
        while (r < len) {
            cur += nums[r];
            if (s.find(nums[r]) == s.end()) {
                s.insert(nums[r]);
            } else {
                while (nums[l] != nums[r]) {
                    cur -= nums[l];
                    s.erase(nums[l]);
                    ++l;
                }
                cur -= nums[l];
                // s.erase(nums[l]);
                ++l;
            }
            ans = max(ans, cur);
            ++r;
        }
        return ans;
    }
};

File: python3/1695.py
from typing import List

class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        s = set()
        ans, cur = 0, 0
        l, r = 0, 0
        n = len(nums)
        while r < n:
            cur += nums[r]
            if nums[r] not in s:
                s.add(nums[r])
            else:
                while nums[l] != nums[r]:
                    cur -= nums[l]
                    s.remove(nums[l])
                    l += 1
                cur -= nums[l]
                # s.remove(nums[l])
                l += 1
            ans = max(ans, cur)
            r += 1
        return ans